bin/milight

```diff
@@ -1,32 +1,199 @@
+  send_command(zones) { zones.hue(options[:color]) } if options[:color]
+  send_command(zones) { zones.saturation(options[:saturation]) } if options[:saturation]
+end
+# rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity,  Metrics/PerceivedComplexity
+
+# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
+def update_zone(zone, options)
+  send_command(zone) { zone.link } if options[:link]
+  send_command(zone) { zone.unlink } if options[:unlink]
+  send_command(zone) { zone.on } if options[:on]
+  send_command(zone) { zone.off } if options[:off]
+  send_command(zone) { zone.night_light } if options[:night]
+  send_command(zone) { zone.brightness(options[:brightness]) } if options[:brightness]
+  send_command(zone) { zone.temperature(options[:temperature]) } if options[:temperature]
+  send_command(zone) { zone.warm_light } if options[:warm]
+  send_command(zone) { zone.white_light } if options[:white]
+  send_command(zone) { zone.hue(options[:hue]) } if options[:hue]
+  send_command(zone) { zone.hue(options[:color]) } if options[:color]
+  send_command(zone) { zone.saturation(options[:saturation]) } if options[:saturation]
 end
+# rubocop:enable Metrics/AbcSize, Metrics/CyclomaticComplexity,  Metrics/PerceivedComplexity, Metrics/MethodLength
 
-host = ARGV.shift
-command = ARGV.shift
-zone = ARGV.shift
+# multiple commands seem to fail when invoked in quick succession,
+# so this function invokes a command (passed in via a block)
+# and then waits a short while
+def send_command(zone)
+  yield
+  zone.wait(0.2)
+end
 
-controller = Milight::V6::Controller.new(host)
+def valid_color?(color, colors)
+  colors.any? color
+end
 
-if zone.nil?
-  lights = controller.all
-else
-  lights = controller.zone(zone.to_i)
+def assert_not_combined(options, keys)
+  raise "can't combine options #{keys}" unless keys.map { |k| options.key?(k) }.filter { |e| e }.length <= 1
 end
 
-case command
-when "link"
-  lights.link
-when "unlink"
-  lights.unlink
-when "off"
-  lights.off
-when "on"
-  lights.on
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength
+def hex_color(value, colors)
+  case value
+  when colors[0]
+    Milight::V6::Color::RED
+  when colors[1]
+    Milight::V6::Color::LAVENDER
+  when colors[2]
+    Milight::V6::Color::BLUE
+  when colors[3]
+    Milight::V6::Color::AQUA
+  when colors[4]
+    Milight::V6::Color::GREEN
+  when colors[5]
+    Milight::V6::Color::LIME
+  when colors[6]
+    Milight::V6::Color::YELLOW
+  when colors[7]
+    Milight::V6::Color::ORANGE
+  else
+    raise "unhandled color #{value}"
+  end
 end
+# rubocop:enable Metrics/CyclomaticComplexity, Metrics/MethodLength
+
+# parse options
+# rubocop:disable Metrics/BlockLength
+options = {}
+usage = "Usage: milight [options]\n"
+OptionParser.new(usage) do |opt|
+  opt.on("-x", "--host HOST", String, "ip address of milight") do |o|
```

Comment:
I think -h would be more logical choice

Reply:
Ah yes - I originally tried this, but I think it's reserved for the help, although, the help only shows with `--help`. I'm not sure how to change that tbh - but there's probably a way